src/pages/SingleProduct.js

```diff
@@ -0,0 +1,149 @@
+import "../styles/single-product.css";
+import axios from "axios";
+import {useParams, Link} from "react-router-dom";
+import {useState, useEffect} from "react";
+import {NavMenu, Footer} from "../components/allComponents";
+import {useCart, useWishList} from "../context/allContext";
+import {isProductInCart, isProductInWishList} from "../utils/allUtils";
+
+const SingleProduct = () => {
+  const params = useParams();
```

Comment:
everything looks fine, addedtocart is unnecessary state, as it can be derived from current state i think, more than 3 use state can be avoided

Reply:
Good catch, updated 👍